068-performance-testing.md

```diff
@@ -0,0 +1,451 @@
+        this.topicOperatorCollector = new TopicOperatorMetricsCollector.Builder()
+                .withScraperPodName(this.testStorage.getScraperPodName())
+                .withNamespaceName(this.testStorage.getNamespaceName())
+                .withComponentType(ComponentType.TopicOperator)
+                .withComponentName(this.testStorage.getClusterName())
+                .build();
+
+        this.topicOperatorPollingThread = new TopicOperatorPollingThread(this.topicOperatorCollector, "strimzi.io/cluster=" + this.testStorage.getClusterName());
+        this.daemonThread = new Thread(this.topicOperatorPollingThread);
+        this.daemonThread.setDaemon(true); // Set as daemon so it doesn't prevent JVM shutdown
+        this.daemonThread.start();
+
+        // 2a. Exercise: Creation of KafkaTopics
+        startTime = System.nanoTime();
+
+        // Create topics
+        KafkaTopicScalabilityUtils.createTopicsViaK8s(testStorage.getNamespaceName(), testStorage.getClusterName(), topicNamePrefix,
+                numberOfTopics, 12, 3, 1);
+
+        KafkaTopicScalabilityUtils.waitForTopicsReady(testStorage.getNamespaceName(), topicNamePrefix, numberOfTopics);
+
+        endTime = System.nanoTime();
+        createTopicsTime = endTime - startTime;
+        LOGGER.info("Time taken to create {} topics: {} ms", numberOfTopics, createTopicsTime);
+
+        endTimeWhole = System.nanoTime();
+        totalTimeWhole = endTimeWhole - startTime;
+
+        LOGGER.info("Time taken to create {} topics and send and recv: {} messages: {} ms", numberOfTopics, NUMBER_OF_MESSAGES, totalTimeWhole);
+        
+        // Start measuring time for deletion of all topics
+        long deletionStartTime = System.nanoTime();
+
+        // 2b. Exercise: Deletion of KafkaTopics
+        ResourceManager.getInstance().deleteResourcesOfType(KafkaTopic.RESOURCE_KIND);
+
+        long deletionEndTime = System.nanoTime();
+        totalDeletionTime = deletionEndTime - deletionStartTime;
+        LOGGER.info("Time taken to delete {} topics: {} ms", numberOfTopics, totalDeletionTime);
+
+        endTimeWhole = System.nanoTime();
+        totalTimeWhole = endTimeWhole - startTime;
+
+        LOGGER.info("Total time taken to create {} topics, send and receive {} messages, and delete topics: {} ms", numberOfTopics, NUMBER_OF_MESSAGES, totalTimeWhole);
+    } finally {
+        // 4a. teardown: Gracefully stop the topic-operator thread
+        // 4b: teardown: Now, it's safe to log performance data as the collection thread has been stopped
+        PerformanceReporter.logPerformanceData(
+                numberOfTopics, numberOfClientInstances, NUMBER_OF_MESSAGES, createTopicsTime, totalSendAndRecvTime, 
+                totalDeletionTime, totalTimeWhole, this.testStorage, ACTUAL_TIME, Environment.PERFORMANCE_DIR, 
+                maxBatchSize, maxBatchLingerMs, this.topicOperatorPollingThread.getMetricsHistory()
+        );
+    }
+}
```

Comment:
I think pasting the whole testcase here is not necessary
Because it's anyway similar code as we have today in STs and everything inside it is just implementation of a test case.

Reply:
Yeah, but this just shows what such tests would look like. It's just for illustrative purposes for a reader to have some clue about:
1. setup with also ThreadCollector ...
2. exercise 
3. verify
4. teardown - performance reporter...

I could delete it or maybe make it shorter